msteams-platform/bots/how-to/conversations/Workflow bot in Teams.md

```diff
@@ -0,0 +1,460 @@
+
+## Card action handler
+
+To simplify the creation of a workflow bot, the TeamsFx SDK provides an Adaptive Card action handler (`TeamsFxAdaptiveCardActionHandler`). You can focus only on the development of business requirement to respond to the card action without learning the Bot Framework.
+
+The following diagram illustrates how to respond to an Adaptive Card action with TeamsFx SDK:
+
+:::image type="content" source="../../../assets/images/sbs-workflow-bot/sbs-workflow-bot-action-card.png" alt-text="workflow bot card action handler diagram" lightbox="../../../assets/images/sbs-workflow-bot/sbs-workflow-bot-action-card.png":::
+
+* **Action card**: The card where you define your action that users can invoke (DoStuff button).
+* **Card action handler**: Triggered when users invoke the corresponding card action (its `triggerVerb` is same as the `verb` property in Adaptive Card action). It can send a response card to respond to the action.
+* **Response card**: The card which respond to the action when user invokes it from the action card.
+
+To handle card actions with TeamsFx SDK, each card action handler must implement `TeamsFxAdaptiveCardActionHandler` interface. This is the interface definition for `TeamsFxAdaptiveCardActionHandler`:
+
+``` Export interface
+
+TeamsFxAdaptiveCardActionHandler 
+{
+    /**
+     * The verb defined in adaptive card action that can trigger this handler.
+     */
+    triggerVerb: string;
+
+    /**
+     * Specify the behavior for how the card response will be sent in Teams conversation.
+     * The default value is `AdaptiveCardResponse.ReplaceForInteractor`, which means the card
+     * response will replace the current one only for the interactor.
+     */
+    adaptiveCardResponse?: AdaptiveCardResponse;
+    
+    /**
+     * The handler function that will be invoked when the action is fired.
+     * @param context The turn context.
+     * @param actionData The contextual data that associated with the action.
+     */
+    handleActionInvoked(context: TurnContext, actionData: any): Promise<InvokeResponse>;
+}
+```
+
+## Customize initialization
+
+The default initialization is located in `bot/src/internal/initialize.js(ts)`.
+
+You can update the initialization logic to:
+
+1. Set `options.adapter` to use your own `BotFrameworkAdapter`.
+1. Set `options.command.commands` to include more command handlers.
+1. Set `options.cardAction.actions` to include more action handlers.
+1. Set `options.{feature}.enabled` to enable more `ConversationBot` functionality.
+
+For more information on initialization customization, see [additional initialization customization](https://github.com/OfficeDev/TeamsFx/wiki/Respond-to-chat-commands-in-Teams#customize-initialization)
+
+## Add card action
```

Comment:
Shall we keep the code snippet for `Add card action` consistent with our workflow bot readme?
https://github.com/OfficeDev/TeamsFx/tree/dev/templates/bot/ts/workflow#extend-the-workflow-bot-template-with-more-actions-and-responses

Reply:
The `Add card action` code snippets were added from technical content provided to us: https://github.com/OfficeDev/TeamsFx/wiki/Respond-to-card-actions-in-Teams#add-more-card-actions
Please confirm if you want us to replace it with all the 4 code snippets in https://github.com/OfficeDev/TeamsFx/tree/dev/templates/bot/ts/workflow#extend-the-workflow-bot-template-with-more-actions-and-responses, respectively?